tools/hlstool/hlstool.cpp

```diff
@@ -119,40 +119,67 @@ static cl::opt<HLSFlow>
                                   "Statically scheduled (Calyx path)")),
             cl::cat(mainCategory));
 
-enum DynamicParallelismKind {
-  DynamicParallelismNone,
-  DynamicParallelismLocking,
-  DynamicParallelismPipelining
-};
+enum DynamicParallelismKind { None, Locking, Pipelining };
 
 static cl::opt<DynamicParallelismKind> dynParallelism(
     "dynamic-parallelism", cl::desc("Specify the DHLS task parallelism kind"),
     cl::values(
-        clEnumValN(DynamicParallelismNone, "none",
+        clEnumValN(None, "none",
                    "Add no protection mechanisms that could prevent data races "
                    "when a function has multiple active invocations."),
-        clEnumValN(DynamicParallelismLocking, "locking",
+        clEnumValN(Locking, "locking",
                    "Add function locking protection mechanism which ensures "
                    "that only one function invocation is active."),
-        clEnumValN(DynamicParallelismPipelining, "pipelining",
+        clEnumValN(Pipelining, "pipelining",
                    "Add function pipelining mechanism that enables a "
                    "pipelined execution of multiple function invocations while "
                    "preserving correctness.")),
-    cl::init(DynamicParallelismPipelining), cl::cat(mainCategory));
+    cl::init(Pipelining), cl::cat(mainCategory));
 
-enum OutputFormatKind { OutputIR, OutputVerilog, OutputSplitVerilog };
+enum IRLevel {
+  // A high-level dialect like affine or scf
+  High,
+  // The IR right before the core lowering dialect
+  PreCompile,
+  // The IR in core dialect
+  Core,
+  // The lowest form of core IR (i.e. after all passes have run)
+  PostCompile,
+  // The IR after lowering is performed
+  RTL,
+  // System verilog representation
+  SV
+};
 
-static cl::opt<int>
-    irInputLevel("ir-input-level",
-                 cl::desc("Level at which to input IR at. It is flow-defined "
-                          "which value corersponds to which IR level."),
-                 cl::init(-1), cl::cat(mainCategory));
+static cl::opt<IRLevel> irInputLevel(
+    "input-level",
+    cl::desc("Level at which to input IR at. It is flow-defined "
+             "which value corersponds to which IR level."),
+    cl::values(clEnumValN(High, "high", "High-level dialects like affine"),
+               clEnumValN(PreCompile, "pre-compile",
+                          "The IR right before the core lowering dialect"),
+               clEnumValN(Core, "core", "The IR in core dialect"),
+               clEnumValN(PostCompile, "post-compile",
+                          "The lowest form of core IR (i.e. after all "
+                          "passes have run)"),
```

Comment:
I think it's a bit ambiguous what "core" is here - is it the "core" dialect of the HLS flow in question (i.e. Calyx for a calyx-based flow) or is the Core RTL dialects (HW/comb/seq)?

Reply:
Done!